smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy

```diff
@@ -907,7 +907,7 @@ def updateSensorData(sensorData) {
 						// is preserved and not changed to name from ecobee cloud as this is the first name change is allowed
 						child.setDisplayName(it.name)
 					}
-					if (!remoteSensors[dni] || remoteSensors[dni].deviceAlive) {
+					if (remoteSensors[dni] && remoteSensors[dni].deviceAlive) {
```

Comment:
you inverted the logic here, right?
before it would have entered the block if it was a new device `!remoteSensor[dni]`, but now it won't

Reply:
Yes as we don't want to add the data for a non selected device even if there's a loitering child hanging around.